superset-frontend/src/views/CRUD/data/dataset/DatasetList.tsx

```diff
@@ -343,7 +343,7 @@ const DatasetList: FunctionComponent<DatasetListProps> = ({
           row: {
             original: { kind },
           },
-        }: any) => kind[0]?.toUpperCase() + kind.slice(1),
+        }: any) => (kind === 'physical' ? t('Physical') : t('Virtual')),
```

Comment:
I suppose this is less flexible, e.g. we'd be in trouble if we ever added another `kind` but I think it's fine for now.